src/playermat/statTracker.ttslua

```diff
@@ -0,0 +1,76 @@
+-- Stat Tracker
+-- made by:         Chr1Z
+-- description:     helps tracking the stats of your investigator
+information = {
+    version = "1.1",
+    last_updated = "12.11.2022"
+}
+
+local BUTTON_PARAMETERS          = {}
+BUTTON_PARAMETERS.function_owner = self
+BUTTON_PARAMETERS.height         = 650
+BUTTON_PARAMETERS.width          = 700
+BUTTON_PARAMETERS.position       = { -4.775, 0.1, -0.03 }
+BUTTON_PARAMETERS.color          = { 0, 0, 0, 0 }
+BUTTON_PARAMETERS.font_color     = { 0, 0, 0, 100 }
+BUTTON_PARAMETERS.font_size      = 450
+X_OFFSET                         = 1.91
+
+function onSave() return JSON.encode(stats) end
+
+-- load stats and make buttons (left to right)
+function onLoad(saved_data)
+    stats = JSON.decode(saved_data) or { 1, 1, 1, 1 }
+
+    for i = 1, 4 do
+        BUTTON_PARAMETERS.position[1] = BUTTON_PARAMETERS.position[1] + X_OFFSET
+        BUTTON_PARAMETERS.label = stats[i] .. "   "
+        BUTTON_PARAMETERS.click_function = attachIndex("button_click", i)
+        self.createButton(BUTTON_PARAMETERS)
+    end
+
+    self.addContextMenuItem("More Information", function()
+        printToAll("------------------------------", "White")
+        printToAll("Stat Tracker v" .. information["version"] .. " by Chr1Z", "Orange")
+        printToAll("last updated: " .. information["last_updated"], "White")
+    end)
+    self.addContextMenuItem("Reset to 1s", function() updateStats({ 1, 1, 1, 1 }) end)
+end
+
+-- helper function to carry index
+function attachIndex(click_function, index)
+    local fn_name = click_function .. index
+    _G[fn_name] = function(obj, player_color, isRightClick)
```

Comment:
I didn't come up with this snippet myself, but I am open for different suggestions!
Maybe this could be solved with local scope but I am not sure how that would look like.